### PR TITLE
Avoid json structures in http

### PR DESCRIPTION
**Pull request by sarmar11**
- Add json functionality to http
- Create Test cases related

### CODE REVIEW

**Review by cbrea — APPROVED** [review]
LGTM

File: steps/http/session.go [hunk 1]
```diff
@@ -284,6 +321,53 @@ func (s *Session) ValidateResponseBodyJSONSchema(ctx context.Context, schema str
...
+	if err = json.Unmarshal([]byte(fmt.Sprint(realResponseBodyJSON)), &realResponseBody); err != nil {
+		return fmt.Errorf("error Unmarshaling response body: %w", err)
+	}
+
+	if !reflect.DeepEqual(jsonResponseBodyMap, realResponseBody) {
+		return fmt.Errorf("expected JSON does not match response JSON, \n%v\n vs \n%s", jsonResponseBodyMap,
+			realResponseBody)
+	}
+
+	return nil
```
Comment (mariolg) [inline steps/http/session.go:368]: This is the same as before comment (refactor).

**Review by mariolg — COMMENTED** [review]
Have a look at my comments.

File: steps/http/json.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+			fmt.Errorf("error unmarshalling JSON at %s due to error: %w", absPath, readErr)
+	}
+
+	for _, response := range dataStruct {
+		if fmt.Sprint(response["code"]) == code {
+			return response[param], nil
+		}
+	}
+
+	return nil, fmt.Errorf("code not found")
```
[the remark this answers is not in the document]
Comment (mariolg) [reply steps/http/json.go]: Resolved in: https://github.com/Telefonica/golium/pull/52/commits/84ebd718378fd2a3f2c28db0a556eeac4929522a

**Review by mariolg — APPROVED** [review]
LGTM

**Pull request merged**